File: 2023/01_Segmentation/Python/functionSaveTIFFMultipage.py
import numpy as np
from PIL import Image

def functionSaveTIFFMultipage(volume, fileNameOutput, bitdepth):
    h, w, d = np.shape(volume)

    if bitdepth == 8:
        volume = np.uint8(volume)
    else:
        volume = np.uint16(volume)

    imlist = []
    #for m in volume:
    for i in range(d):
        #imlist.append(Image.fromarray(m))
        m = volume[:,:,i]
        imlist.append(Image.fromarray(m))

    imlist[0].save(fileNameOutput, compression="tiff_lzw", save_all=True,
                   append_images=imlist[1:])

File: 2023/01_Segmentation/MATLAB/aux_functions/functionSaveTIFFMultipage.m
function functionSaveTIFFMultipage(volume,filenameOutput,bitdepth)
[h,w,d]=size(volume);

if bitdepth == 8
    volume = uint8(volume);
else
    volume = uint16(volume);
end

imwrite(volume(:,:,1), filenameOutput);
for i=2:d
    imwrite(volume(:,:,i), filenameOutput, 'writemode', 'append');
end


end

